huaweicloud/services/acceptance/identitycenter/data_source_huaweicloud_identitycenter_users_test.go

```diff
@@ -0,0 +1,96 @@
+package identitycenter
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance"
+)
+
+func TestAccDatasourceIdentityCenterUsers_basic(t *testing.T) {
+	name := acceptance.RandomAccResourceName()
+	rName := "data.huaweicloud_identitycenter_users.test"
+	dc := acceptance.InitDataSourceCheck(rName)
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:          func() { acceptance.TestAccPreCheck(t) },
+		ProviderFactories: acceptance.TestAccProviderFactories,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccDatasourceIdentityCenterUsers_basic(name),
+				Check: resource.ComposeTestCheckFunc(
+					dc.CheckResourceExists(),
+					resource.TestCheckResourceAttrSet(rName, "users.0.id"),
+					resource.TestCheckResourceAttrSet(rName, "users.0.user_name"),
+					resource.TestCheckResourceAttrSet(rName, "users.0.family_name"),
+					resource.TestCheckResourceAttrSet(rName, "users.0.given_name"),
+					resource.TestCheckResourceAttrSet(rName, "users.0.display_name"),
+					resource.TestCheckResourceAttrSet(rName, "users.0.email"),
+					resource.TestCheckOutput("user_name_filter_is_useful", "true"),
+					resource.TestCheckOutput("family_name_filter_is_useful", "true"),
+					resource.TestCheckOutput("given_name_filter_is_useful", "true"),
+					resource.TestCheckOutput("display_name_filter_is_useful", "true"),
+					resource.TestCheckOutput("email_filter_is_useful", "true"),
+				),
+			},
+		},
+	})
+}
+
+func testAccDatasourceIdentityCenterUsers_basic(name string) string {
+	return fmt.Sprintf(`
+%s
+
+data "huaweicloud_identitycenter_users" "test" {
+  identity_store_id = data.huaweicloud_identitycenter_instance.test.identity_store_id
+  user_name         = huaweicloud_identitycenter_user.test.user_name
+}
+
+data "huaweicloud_identitycenter_users" "user_name_filter" {
+  identity_store_id = data.huaweicloud_identitycenter_instance.test.identity_store_id
+  user_name         = huaweicloud_identitycenter_user.test.user_name
+}
+output "user_name_filter_is_useful" {
+  value = alltrue([for v in data.huaweicloud_identitycenter_users.user_name_filter.users[*].user_name:
```

Comment:
it will fail when the filtering users is empty, please refer to the acceptance of huaweicloud_dns_zones

Reply:
fixed